test.e2e/spec/services.spec.js

```diff
@@ -333,6 +333,43 @@ define(['angular', 'given', 'util'], function(angular, given, util) {
       });
     });
 
+    describe('$resource for model with custom scope-like methods', function() {
+      var $injector;
+      before(function() {
+        return given.servicesForLoopBackApp(
+          {
+            models: {
+              'pretender': {},
+            },
+            setupFn: (function(app, cb) {
+              var Pretender = app.models.Pretender;
+              Pretender.prototype['__get__spaces'] = function(cb) {
+                cb(null, [1,2,3]);
+              };
+              Pretender.remoteMethod(
+                '__get__spaces',
+                {
+                  isStatic: false,
+                  http: { path: '/spaces', verb: 'get' },
+                  returns: { arg: 'spaces', type: 'array', root: true }
+                }
+              );
+              cb();
+            }).toString(),
+          })
+          .then(function(createInjector) {
+            $injector = createInjector();
+          });
+      });
+
+      it('has a client method generated', function() {
+        var Pretender = $injector.get('Pretender');
+        var methodNames = Object.keys(Pretender);
+        console.log('methods', methodNames);
```

Comment:
Thank you for reviewing the patch!

This log is actually intended. When the assertion on the following line fails, it unfortunately does not print the contents of `methodNames`, which makes it difficult to troubleshoot.